### PR TITLE
Colin/utxo refactor

### PR DESCRIPTION
**Pull request by colin-axner**
Major refactor of tests. 
name changing in SpendMsg to match with rootchain contract

### CODE REVIEW

File: types/utxo.go [hunk 1]
```diff
@@ -33,65 +45,92 @@ func NewBaseUTXO(addr common.Address, inputaddr [2]common.Address, amount uint64
...
+	}
+	baseutxo.Position = plasmaposition
+	return nil
+}
+
+func (baseutxo BaseUTXO) GetDenom() string {
+	return "Ether"
+}
+
+func (baseutxo *BaseUTXO) SetDenom(denom string) error {
```
Comment (AdityaSripal) [inline types/utxo.go]: SetDenom should throw error here

File: x/utxo/types.go [hunk 2]
```diff
@@ -23,22 +23,20 @@ type UTXO interface {
 // Positions must be unqiue or a collision may result when using mapper.go
 type Position interface {
 	// Position is a uint slice
-	Get()         []uint64 // get position int slice. Return nil if unset.
-	Set([]uint64) error    // errors if already set
-	// Used to convert Position to Bytes to store in mapper
-	Bytes()       []byte 
+	Get() []uint64 // get position int slice. Return nil if unset.
```
Comment (AdityaSripal) [inline x/utxo/types.go]: On second thought, this should be []big.Int or []sdk.Int

File: types/utxo.go (hunk 1 above)
Comment (colin-axner) [reply types/utxo.go]: handler in x/utxo needs to be able to set the Denom

Comment (AdityaSripal) [reply types/utxo.go]: Sorry SetDenom should just do nothing, just return nil

**Pull request merged**